File: src/main/java/net/ccbluex/liquidbounce/ui/client/GuiAntiForge.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.ui.client;

import lol.liquidcat.LiquidCat;
import net.ccbluex.liquidbounce.features.special.AntiForge;
import net.ccbluex.liquidbounce.ui.font.Fonts;
import net.minecraft.client.gui.GuiButton;
import net.minecraft.client.gui.GuiScreen;
import org.lwjgl.input.Keyboard;

import java.io.IOException;

public class GuiAntiForge extends GuiScreen {

    private final GuiScreen prevGui;

    private GuiButton enabledButton;
    private GuiButton fmlButton;
    private GuiButton proxyButton;
    private GuiButton payloadButton;

    public GuiAntiForge(final GuiScreen prevGui) {
        this.prevGui = prevGui;
    }

    @Override
    public void initGui() {
        buttonList.add(enabledButton = new GuiButton(1, width / 2 - 100, height / 4 + 35, "Enabled (" + (AntiForge.enabled ? "On" : "Off") + ")"));
        buttonList.add(fmlButton = new GuiButton(2, width / 2 - 100, height / 4 + 50 + 25, "Block FML (" + (AntiForge.blockFML ? "On" : "Off") + ")"));
        buttonList.add(proxyButton = new GuiButton(3, width / 2 - 100, height / 4 + 50 + 25 * 2, "Block FML Proxy Packet (" + (AntiForge.blockProxyPacket ? "On" : "Off") + ")"));
        buttonList.add(payloadButton = new GuiButton(4, width / 2 - 100, height / 4 + 50 + 25 * 3, "Block Payload Packets (" + (AntiForge.blockPayloadPackets ? "On" : "Off") + ")"));

        buttonList.add(new GuiButton(0, width / 2 - 100, height / 4 + 55 + 25 * 4 + 5, "Back"));
    }

    @Override
    protected void actionPerformed(GuiButton button) {
        switch(button.id) {
            case 1:
                AntiForge.enabled = !AntiForge.enabled;
                enabledButton.displayString = "Enabled (" + (AntiForge.enabled ? "On" : "Off") + ")";
                LiquidCat.fileManager.saveConfig(LiquidCat.fileManager.valuesConfig);
                break;
            case 2:
                AntiForge.blockFML = !AntiForge.blockFML;
                fmlButton.displayString = "Block FML (" + (AntiForge.blockFML ? "On" : "Off") + ")";
                LiquidCat.fileManager.saveConfig(LiquidCat.fileManager.valuesConfig);
                break;
            case 3:
                AntiForge.blockProxyPacket = !AntiForge.blockProxyPacket;
                proxyButton.displayString = "Block FML Proxy Packet (" + (AntiForge.blockProxyPacket ? "On" : "Off") + ")";
                LiquidCat.fileManager.saveConfig(LiquidCat.fileManager.valuesConfig);
                break;
            case 4:
                AntiForge.blockPayloadPackets = !AntiForge.blockPayloadPackets;
                payloadButton.displayString = "Block Payload Packets (" + (AntiForge.blockPayloadPackets ? "On" : "Off") + ")";
                LiquidCat.fileManager.saveConfig(LiquidCat.fileManager.valuesConfig);
                break;
            case 0:
                mc.displayGuiScreen(prevGui);
                break;
        }
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        drawBackground(0);
        Fonts.fontBold180.drawCenteredString("AntiForge", (int) (width / 2F), (int) (height / 8F + 5F), 4673984, true);

        super.drawScreen(mouseX, mouseY, partialTicks);
    }

    @Override
    protected void keyTyped(char typedChar, int keyCode) throws IOException {
        if(Keyboard.KEY_ESCAPE == keyCode) {
            mc.displayGuiScreen(prevGui);
            return;
        }

        super.keyTyped(typedChar, keyCode);
    }
}

File: src/main/kotlin/lol/liquidcat/ui/client/guis/GuiAntiForge.kt
/*
 * LiquidCat Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CatsDevelopment/LiquidCat
 */
package lol.liquidcat.ui.client.guis

import lol.liquidcat.features.misc.AntiForge
import lol.liquidcat.file.FileManager.saveConfig
import lol.liquidcat.file.FileManager.valuesConfig
import net.ccbluex.liquidbounce.ui.font.Fonts
import net.minecraft.client.gui.GuiButton
import net.minecraft.client.gui.GuiScreen
import org.lwjgl.input.Keyboard

class GuiAntiForge(private val prevGui: GuiScreen) : GuiScreen() {

    private lateinit var enabledButton: GuiButton
    private lateinit var fmlButton: GuiButton
    private lateinit var proxyButton: GuiButton
    private lateinit var payloadButton: GuiButton

    override fun initGui() {
        val dWidth = width / 2 - 100
        val dHeight = height / 4 + 50

        buttonList.add(GuiButton(1, dWidth, dHeight - 15, "Enabled (${if (AntiForge.enabled) "On" else "Off"})").also { enabledButton = it })
        buttonList.add(GuiButton(2, dWidth, dHeight + 25, "Block FML (${if (AntiForge.blockFML) "On" else "Off"})").also { fmlButton = it })
        buttonList.add(GuiButton(3, dWidth, dHeight + 25 * 2, "Block FML Proxy Packet (${if (AntiForge.blockProxyPacket) "On" else "Off"})").also { proxyButton = it })
        buttonList.add(GuiButton(4, dWidth, dHeight + 25 * 3, "Block Payload Packets (${if (AntiForge.blockPayloadPackets) "On" else "Off"})").also { payloadButton = it })
        buttonList.add(GuiButton(0, dWidth, dHeight + 5 + 25 * 4 + 5, "Back"))
    }

    override fun actionPerformed(button: GuiButton) {
        when (button.id) {
            0 -> mc.displayGuiScreen(prevGui)
            1 -> {
                AntiForge.enabled = !AntiForge.enabled
                enabledButton.displayString = "Enabled (${if (AntiForge.enabled) "On" else "Off"})"
                saveConfig(valuesConfig, false)
            }
            2 -> {
                AntiForge.blockFML = !AntiForge.blockFML
                fmlButton.displayString = "Block FML (${if (AntiForge.blockFML) "On" else "Off"})"
                saveConfig(valuesConfig, false)
            }
            3 -> {
                AntiForge.blockProxyPacket = !AntiForge.blockProxyPacket
                proxyButton.displayString = "Block FML Proxy Packet (${if (AntiForge.blockProxyPacket) "On" else "Off"})"
                saveConfig(valuesConfig, false)
            }
            4 -> {
                AntiForge.blockPayloadPackets = !AntiForge.blockPayloadPackets
                payloadButton.displayString = "Block Payload Packets (${if (AntiForge.blockPayloadPackets) "On" else "Off"})"
                saveConfig(valuesConfig, false)
            }
        }
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        drawBackground(0)
        Fonts.fontBold180.drawCenteredString(
            "AntiForge",
            (width / 2f).toInt().toFloat(),
            (height / 8f + 5f).toInt().toFloat(),
            4673984,
            true
        )
        super.drawScreen(mouseX, mouseY, partialTicks)
    }

    override fun keyTyped(typedChar: Char, keyCode: Int) {
        if (Keyboard.KEY_ESCAPE == keyCode) {
            mc.displayGuiScreen(prevGui)
            return
        }
        super.keyTyped(typedChar, keyCode)
    }
}